docs/external-node.md

```diff
@@ -0,0 +1,577 @@
+  - [Bypass Antrea NetworkPolicy](#bypass-antrea-networkpolicy)
+- [OpenFlow pipeline](#openflow-pipeline)
+  - [Non-IP packet](#non-ip-packet)
+  - [IP packet](#ip-packet)
+- [Limitations](#limitations)
+<!-- /toc -->
+
+## What is ExternalNode?
+
+`ExternalNode` is a CRD API that enables Antrea to manage the network connectivity
+and security on a Non-Kubernetes Node (like a virtual machine or a bare-metal
+server). It supports specifying which network interfaces on the external Node
+are expected to be protected with Antrea NetworkPolicy rules. The virtual machine
+or bare-metal server represented by an `ExternalNode` resource can be either
+Linux or Windows. An "external Node" will be used to represent such a virtual
+machine or a bare-metal server in this document.
+
+Antrea NetworkPolicies are applied to an external Node by leveraging the
+`ExternalEntity` resource. `antrea-controller` creates an `ExternalEntity`
+resource for each network interface defined in the `ExternalNode` resource.
+
+`antrea-agent` is running on the external Node, and it controls network
+connectivity and security by attaching the network interface(s) to an OVS bridge.
+A new OpenFlow pipeline has been implemented, dedicated to the ExternalNode
+feature.
+
+You may be interested in using this capability for the below scenarios:
+
+- To apply Antrea NetworkPolicy to an external Node.
+- You want the same security configurations on the external Node for all
+  Operating Systems.
+
+This guide demonstrates how to configure `ExternalNode` to achieve the above
+result.
+
+## Prerequisites
+
+`ExternalNode` is introduced in v1.8 as an alpha feature. The feature gate
+`ExternalNode` must be enabled in the `antrea-controller` and `antrea-agent`
+configuration. The configuration for `antrea-controller` is modified in the
+`antrea-config` ConfigMap as follows for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+The `antrea-controller` implements the `antrea` Service, which accepts
+connections from each `antrea-agent` and is an important part of the
+NetworkPolicy implementation. By default, the `antrea` Service has type
+`ClusterIP`. Because external Nodes run outside of the Kubernetes cluster, they
+cannot directly access the `ClusterIP` address. Therefore, the `antrea` Service
+needs to become externally-accessible, by changing its type to `NodePort` or
+`LoadBalancer`.
+
+Since `antrea-agent` is running on an external Node which is not managed by
+Kubernetes, a configuration file needs to be present on each machine where the
+`antrea-agent` is running, and the path to this file will be provided to the
+`antrea-agent` as a command-line argument.
+
+Refer to the [sample configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
```

Comment:
Since the next sentence is a referrence to the installation part, it is a bit strange to put it before this one. Do you think it is better if I remove the configuration referrence sentence?

Reply:
Sorry, I did not express clearly. I meant merge this sentence to the previous paragraph (line 88).

Reply:
Got it, updated.